src/patterns.rs

```diff
@@ -55,39 +65,105 @@ impl Rewrite for Pat {
+                match *pat_vec {
+                    Some(ref pat_vec) => {
+                        if pat_vec.is_empty() {
+                            Some(path_str)
+                        } else {
+                            // 1 = (
+                            let width = try_opt!(width.checked_sub(path_str.len() + 1));
+                            let offset = offset + path_str.len() + 1;
+                            let items = itemize_list(context.codemap,
+                                                     pat_vec.iter(),
+                                                     ")",
+                                                     |item| item.span.lo,
+                                                     |item| item.span.hi,
+                                                     |item| item.rewrite(context, width, offset),
+                                                     span_after(self.span, "(", context.codemap),
+                                                     self.span.hi);
+                            Some(format!("{}({})",
+                                         path_str,
+                                         try_opt!(format_item_list(items,
+                                                                   width,
+                                                                   offset,
+                                                                   context.config))))
+                        }
+                    }
+                    None => Some(format!("{}(..)", path_str)),
                 }
             }
             Pat_::PatLit(ref expr) => expr.rewrite(context, width, offset),
-            // FIXME(#8): format remaining pattern variants.
-            Pat_::PatIdent(_, _, Some(..)) |
-            Pat_::PatEnum(_, None) |
-            Pat_::PatStruct(..) |
-            Pat_::PatVec(..) |
+            Pat_::PatVec(ref prefix, ref slice_pat, ref suffix) => {
+                // Rewrite all the sub-patterns.
+                let prefix = prefix.iter().map(|p| p.rewrite(context, width, offset));
+                let slice_pat = slice_pat.as_ref().map(|p| {
+                    Some(format!("{}..", try_opt!(p.rewrite(context, width, offset))))
+                });
+                let suffix = suffix.iter().map(|p| p.rewrite(context, width, offset));
+
+                // Munge them together.
+                let pats = prefix.chain(slice_pat.into_iter()).chain(suffix);
+
+                // Check that all the rewrites succeeded, and if not return None.
+                let (somes, nones) = pats.partition::<Vec<Option<String>>, _>(Option::is_some);
+                if nones.len() > 0 {
+                    return None;
+                }
+
+                // Unwrap all the sub-strings and join them with commas.
+                let pats = somes.into_iter().map(|p| p.unwrap()).collect::<Vec<_>>().join(", ");
+                Some(format!("[{}]", pats))
```

Comment:
Are we sure we're staying within the bounds here? If not, we should call `wrap_str`.